cmd/bucket-handlers-listobjects.go

```diff
@@ -81,6 +82,12 @@ func (api objectAPIHandlers) ListObjectsV2Handler(w http.ResponseWriter, r *http
 		marker = startAfter
 	}
 
+	if marker == "" {
```

Comment:
After discussing your change with @kannappanr, it looks like this part of the code will never be executed due to the logic in the previous code block which simply sets the marker if the continuation token is `""`.
```
 marker := token
 // Check if 'continuation-token' is empty.
 if token == "" {
 	// Then we need to use 'start-after' as marker instead.
 	marker = startAfter
 }
```
So,  marker will never be empty  and the error will never be thrown.